custom_components/climate_template/climate.py

```diff
@@ -604,8 +636,9 @@ def _update_preset_mode(self, preset_mode):
 
     def _update_fan_mode(self, fan_mode):
         if fan_mode in self._attr_fan_modes:
-            self._current_fan_mode = fan_mode
-            self.hass.async_create_task(self.async_set_fan_mode(fan_mode))
```

Comment:
@jcwillox This is to avoid triggering the same action again when the value is changed only internally (templated value). Previously this would lead to first the template value changing i.e. the fan mode on the actual device due to a real world change made by the user or by a Home Assistant automation which would then again trigger the same action i.e. set the same state already set by the user or an automation. So basically if the fan_mode of climate.ac changes where the fan_mode_template is:

```
fan_mode_template: "{{ state_attr('climate.ac', 'fan_mode') }}"
```

and set_fan_mode is:
```
set_fan_mode:
  - service: climate.set_fan_mode
    data:
      entity_id: climate.ac
      fan_mode: "{{ fan_mode }}"
```

Only the fan mode of the templated entity will be updated to reflect the new fan mode and set_fan_mode won't be called again to set the same state again resulting in a loop of state changes.

This fixes https://github.com/jcwillox/hass-template-climate/issues/79

This also fixed my climate.ac (which is an IR remote for my Air Conditioner added to Home Assistant through Homebridge) from turning to 'cool' state at every restart even if the previous state was off when it was being templated by this integration.

Reply:
Well that's a good enough reason, thanks for explaining, I'm happy with the change in functionality then. If anyone needs the old functionality they can always use an automation to sync the state via the set actions.